### DISCUSSION TITLE
Possible change in dedection script

### DISCUSSION THREAD

**Discussion by supajo** [svrooij/WinTuner#141] [Ideas]
Hello,

i would like to ask for your opinion on the detection script and whether it would be possible to mark a Win32 app as installed if there is more than one version on the system. For instance, IntelliJ does not update itself but installs alongside the old version. The detection script identifies the old version and returns it, ignoring the new version.

Part of $Wingetoutput:

```
Name                   ID                              Version  Quelle
-----------------------------------------------------------------------
IntelliJ IDEA 2024.3   JetBrains.IntelliJIDEA.Ultimate 2024.3   winget
IntelliJ IDEA 2024.1.4 JetBrains.IntelliJIDEA.Ultimate 2024.1.4 winget
```

I'm thinking about something like this :)

```
    if ($hasTargetVersion -and $hasOlderVersion) {
        Write-Host "Target version ($version) and an older version are installed."
        Exit 0
```

**Comment by svrooij** [comment]
The detection script is configured as always detect this version or higher. Meaning an older version is not detected by the new version package.

But an older version will still report as installed even if they were updated

**Reply by supajo** [reply]
In the case of IntelliJ, Intune gave me back "The application was not detected after installation completed successfully (0x87D1041C)"
And according to the dedection script i got 

"JetBrains.IntelliJIDEA.Ultimate version 2024.1.4 is installed"
"Installed version is lower"

But it never says 2024.3 is installed even if $Wingetoutput has the information
